e2e/services/navigation.service.ts

```diff
@@ -0,0 +1,63 @@
+import { browser, ExpectedConditions as EC } from 'protractor';
+import { LoginPage } from '../page-objects/login/login.po';
+import { MainPage } from '../page-objects/main/main.po';
+import { BasePage } from '../page-objects/base-page';
+import { TestUtil } from '../utilities/test-util';
+import { ButtonState, FormService } from './form.service';
+
+declare const allure: any;
+
+export class NavigationService {
+
+    public static readonly TIME_OUT_MS_FOR_DIALOG_WINDOW = 30000;
+
+    public static login(loginPage: LoginPage, useWrongPassword: boolean = false, clickEnter: boolean = true) {
+        loginPage.getUsernameField().setText(browser.params.login.user);
+        if (!useWrongPassword) {
+            loginPage.getPasswordField().setText(browser.params.login.password);
+        } else {
+            loginPage.getPasswordField().setText('PassProvidedByTesting');
+        }
+        if (clickEnter) {
+            loginPage.getEnterButton().click();
+        }
+    }
+
+    public static navigateToHomePage(loginPage: LoginPage) {
+        browser.get('/#/login');
+        loginPage.retrieveAppParams(loginPage);
+    }
+
+    public static navigateToPatientMaintenancePage(mainPage: MainPage) {
+        this.checkPageIsPresentAndDisplayed(mainPage);
+        mainPage.getPatientButton().click();
+    }
+
+    public static navigateToAllergyMaintenancePage(mainPage: MainPage) {
+        this.checkPageIsPresentAndDisplayed(mainPage);
+        mainPage.getConfigIcon().click();
+    }
+
+    public static waitForPageToBeVisible(page: BasePage) {
+        browser.wait(EC.presenceOf(page.getElementFinder()), NavigationService.TIME_OUT_MS_FOR_DIALOG_WINDOW, 'Main Dialog Window is taking too long to appear in the DOM (timeout: ' + NavigationService.TIME_OUT_MS_FOR_DIALOG_WINDOW + ' ms).');
+    }
+
+    public static checkPageIsPresentAndDisplayed(page: BasePage) {
```

Comment:
The navigation service does not contain expects, I recommend to move the following 2 methods to another service.  According to that, this service will just contain methods to navigate.

Reply:
I will move it